compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -204,7 +204,7 @@ void TR_X86ProcessorInfo::initialize(TR::CodeGenerator *cg)
 
 
 void
-OMR::X86::CodeGenerator::initialize(TR::Compilation *comp)
+OMR::X86::CodeGenerator::initializeX86(TR::Compilation *comp)
```

Comment:
Why is this different than the rest?

Reply:
This API already existed in the X86 CodeGenerator constructor prior to this PR.  I renamed it to `initializeX86` to distinguish it from the name of the `initialize` function name I want to use for the purposes of the CodeGenerator constructor factory.

`initializeX86` (as it is now called) was introduced at some point in the past to common up some initialization between the AMD64 and IA32 code generators.  It is called from somewhere in the middle of the current constructors.  I'm not sure why the constructors weren't simply modified to do this more naturally, but that's the state of the world at the moment.

I think a follow-on to this factory work is to change the x86 CodeGenerator hierarchy to initialize itself using the `initialize()` functions introduced by this PR.  This might need some careful thinking and inspection as there are likely ordering requirements that would change given where `initializeX86` is presently being called.  I will open a follow-up issue for it.

Reply:
Ok, thanks for explaining. I was wondering why we couldn't just push this function into the common x86 `CodeGenerator` from which we derive the `ADM64` and `i386` versions. The ordering explains why that may not be possible or needs to be thought about. Thanks!